cloud/ec2.go

```diff
@@ -349,6 +342,67 @@ func (m *ec2Manager) spawnSpotHost(ctx context.Context, h *host.Host, ec2Setting
 	return resources, nil
 }
 
+// trySpawningSpotHost attempts to spawn a spot host in multiple availability regions. It returns an error if it fails.
+func (m *ec2Manager) trySpawningSpotHost(ctx context.Context, h *host.Host, ec2Settings *EC2ProviderSettings, blockDevices []*ec2.BlockDeviceMapping, osName osType, onDemandPrice float64) ([]*string, error) {
+	azSpotPrices, err := pkgCachingPriceFetcher.getLatestSpotCostsForInstance(ctx, m.client, ec2Settings, osName)
+	if err != nil {
+		return nil, errors.Wrap(err, "error getting latest spot prices")
+	}
+
+	if len(azSpotPrices) == 0 {
+		return nil, errors.New("could not fetch spot host prices for availability zones")
+	}
+
+	catcher := grip.NewBasicCatcher()
+
+	if m.provider == autoProvider {
+		ec2Settings.BidPrice = onDemandPrice
+	}
+
+	originalZone := h.Zone
+	originalSubnetID := ec2Settings.SubnetId
+
+	// Try spawning spot hosts until either one is spawned or all availability zones have been tried.
+	// For auto mode, also stop trying to spawn a spot host early if the on-demand price is cheaper.
+	for _, azSpotPrice := range azSpotPrices {
+		if m.provider == autoProvider && azSpotPrice.price > onDemandPrice {
+			break
+		}
+
+		h.ComputeCostPerHour = azSpotPrice.price
+		h.Zone = azSpotPrice.availabilityZone
+
+		if ec2Settings.VpcName != "" {
+			subnetID, err := m.getSubnetForAZ(ctx, azSpotPrice.availabilityZone, ec2Settings.VpcName)
+			if err != nil {
+				catcher.Add(err)
+				continue
+			}
+			ec2Settings.SubnetId = subnetID
+		}
+
+		resources, err := m.spawnSpotHost(ctx, h, ec2Settings, blockDevices)
```

Comment:
Will this actually fail to spawn a spot host at this point? Spawning a spot host makes a request for a host, but I'm not sure if we will know at this point if there isn't capacity available. We may instead have to do something in the set-cloud-hosts-ready job instead, in order to indicate (via the event log?) that we weren't able to get a host. Another alternative is seeing if there's a different API call, like the spot fleet request. I think Brian McCarthy may have some experience with that API call.